src/gui/win_nwg_ext/mod.rs

```diff
@@ -167,3 +170,69 @@ impl MenuItemEx for nwg::MenuItem {
         }
     }
 }
+
+pub trait WindowEx {
+    fn set_position_ex(&self, x: i32, y: i32);
+}
+pub fn dpi() -> i32 {
+    // prevents GDI DC resource leak
+    use winapi::um::wingdi::GetDeviceCaps;
+    use winapi::um::wingdi::LOGPIXELSX;
+    use winapi::um::winuser::{GetDC, ReleaseDC};
+    let screen = unsafe { GetDC(std::ptr::null_mut()) };
+    let dpi = unsafe { GetDeviceCaps(screen, LOGPIXELSX) };
+    let _ = unsafe { ReleaseDC(std::ptr::null_mut(), screen) };
+    dpi
+}
+pub fn logical_to_physical(x: i32, y: i32) -> (i32, i32) {
+    use muldiv::MulDiv;
+    use winapi::um::winuser::USER_DEFAULT_SCREEN_DPI;
+    let dpi = dpi();
+    let x = x.mul_div_round(dpi, USER_DEFAULT_SCREEN_DPI).unwrap_or(x);
+    let y = y.mul_div_round(dpi, USER_DEFAULT_SCREEN_DPI).unwrap_or(y);
+    (x, y)
+}
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
Clippy suggestion here is correct and the public function should be unsafe and have some Safety doc. Well, the winapi call will probably 

```suggestion
/// # Safety
/// The `handle` param must be a valid pointer to a window handle returned by some winapi call.
/// Failure to do so probably won't be UB because the handle is passed to a WinAPI call
/// which is expected to handle these cases safely, but seems worth noting anyway.
```